tests/by-util/test_more.rs

```diff
@@ -126,3 +126,27 @@ fn test_invalid_file_perms() {
             .stderr_contains("permission denied");
     }
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_more_non_utf8_paths() {
+    use std::fs;
+
+    if std::io::stdout().is_terminal() {
+        let (at, mut ucmd) = at_and_ucmd!();
+
+        // Create test file with normal name first
+        at.write("temp.txt", "test content for non-UTF-8 file");
+
+        // Rename to non-UTF-8 name
+        #[cfg(unix)]
+        {
+            use std::os::unix::ffi::OsStrExt;
+            let file_name = std::ffi::OsStr::from_bytes(b"test_\xFF\xFE.txt");
+            fs::rename(at.subdir.join("temp.txt"), at.subdir.join(file_name)).unwrap();
```

Comment:
I don't understand why you rename the file instead of creating the file with the non-UTF-8 chars directly.